airflow/hooks/python_virtualenv.py

```diff
@@ -0,0 +1,350 @@
+from airflow.models.variable import Variable
+from airflow.utils import hashlib_wrapper
+from airflow.utils.python_virtualenv import prepare_virtualenv
+
+if TYPE_CHECKING:
+    from airflow.typing_compat import Literal
+
+log = logging.getLogger(__name__)
+
+
+class PythonVirtualenvHook(BaseHook):
+    """
+    To run a function in a virtualenv, create virtualenv first and destroy automatically.
+
+    :param venv_conn_id: The connection id to get the virtualenv configuration from.
+    :param requirements: Either a list of requirement strings, or a (templated)
+        "requirements file" as specified by pip.
+    :param python_version: The Python version to run the virtual environment with. Note that
+        both 2 and 2.7 are acceptable forms.
+    :param serializer: Which serializer use to serialize the args and result. It can be one of the following:
+
+        - ``"pickle"``: (default) Use pickle for serialization. Included in the Python Standard Library.
+        - ``"cloudpickle"``: Use cloudpickle for serialize more complex types,
+          this requires to include cloudpickle in your requirements.
+        - ``"dill"``: Use dill for serialize more complex types,
+          this requires to include dill in your requirements.
+    :param system_site_packages: Whether to include
+        system_site_packages in your virtual environment.
+        See virtualenv documentation for more information.
+    :param pip_install_options: a list of pip install options when installing requirements
+        See 'pip install -h' for available options
+    :param index_urls: an optional list of index urls to load Python packages from.
+        If not provided the system pip conf will be used to source packages from.
+    :param venv_cache_path: Optional path to the virtual environment parent folder in which the
+        virtual environment will be cached, creates a sub-folder venv-{hash} whereas hash will be replaced
+        with a checksum of requirements. If not provided the virtual environment will be created and deleted
+        in a temp folder for every execution.
+    """
+
+    conn_name_attr = "venv_conn_id"
+    default_conn_name = "venv_default"
+    conn_type = "python_venv"
+    hook_name = "Python Virtualenv"
+
+    @classmethod
+    def get_ui_field_behaviour(cls) -> dict[str, Any]:
+        v = sys.version_info
+
+        return {
+            "hidden_fields": ["host", "schema", "port", "login", "password", "extra"],
+            "relabeling": {},
+            "placeholders": {"python_version": f"{v.major}.{v.minor}.{v.micro}", "serializer": "pickle"},
+        }
+
+    @classmethod
+    def get_connection_form_widgets(cls) -> dict[str, Any]:
+        from flask_appbuilder.fieldwidgets import BS3TextAreaFieldWidget, BS3TextFieldWidget
+        from flask_babel import lazy_gettext
+        from wtforms import BooleanField, StringField
+        from wtforms.validators import InputRequired, Optional, any_of
```

Comment:
```suggestion
        from wtforms.validators import InputRequired, Optional, AnyOf
```
이번에 개발하면서 느낀건데 훅쪽에서 Alias 쓰는 규칙을 잘 모르겠더라구요 ㅋㅋ 뭐는 클래스처럼쓰고 뭐는 함수처럼쓰고

Reply:
![image](https://github.com/user-attachments/assets/c47b5952-a6d6-454a-a6e3-63757fad07bd)
ㅋㅋㅋㅋ

Reply:
@softyoungha 
얘네 코드에도 보면 거의 `any_of`를 많이쓰더라구요 ㅋㅋ
머가 맞는 컨벤션인지 판단이 안섰습니다 ㅋㅋ